### PR TITLE
[#12] Establish connection with network

### PR DESCRIPTION
**Pull request by szerepak**
Work in progress, nodes are reacting to ping but somehow they do not to [getaddr](https://en.bitcoin.it/wiki/Protocol_documentation#getaddr) command.

### CODE REVIEW

File: clove/network/base.py [hunk 1]
```diff
@@ -21,16 +25,30 @@ def get_nodes(seed) -> list:
...
+                        continue
+                    return
+
+    def version_packet(self):
+        packet = msg_version()
+        packet.addrFrom.ip, packet.addrFrom.port = self.connection.getsockname()
+        packet.addrTo.ip, packet.addrTo.port = self.connection.getpeername()
+        return packet.to_bytes()
+
+    def ping(self):
```
Comment (trojkat) [inline clove/network/base.py]: unused?

File: clove/network/base.py [hunk 2]
```diff
@@ -21,16 +25,30 @@ def get_nodes(seed) -> list:
...
+        if self.connection is None:
+            for seed in self.seeds:
+                nodes = self.get_nodes(seed)
+                for node in nodes:
+                    try:
+                        self.connection = socket.create_connection(
+                            address=(node, self.port),
+                            timeout=timeout
+                        )
+                        self.connection.send(self.version_packet())
```
Comment (trojkat) [inline clove/network/base.py:41]: No `recv`?

File: clove/network/base.py [hunk 3]
```diff
@@ -21,16 +25,30 @@ def get_nodes(seed) -> list:
...
+                        self.connection.send(msg_verack().to_bytes())
```
Comment (trojkat) [inline clove/network/base.py]: We will need `recv` again here.

Comment (szerepak) [reply clove/network/base.py]: what for?

File: clove/network/base.py (hunk 1 above)
Comment (szerepak) [reply clove/network/base.py]: Unused, but useful. You think, I should remove it, no?

File: clove/network/base.py [hunk 4]
```diff
@@ -21,16 +28,49 @@ def get_nodes(seed) -> list:
...
+            for seed in self.seeds:
+                nodes = self.get_nodes(seed)
+                for node in nodes:
+                    try:
+                        self.connection = socket.create_connection(
+                            address=(node, self.port),
+                            timeout=timeout
+                        )
+                        self.connection.send(self.version_packet())
+                        sleep(0.1)
```
Comment (trojkat) [inline clove/network/base.py:42]: Just for the record: try to use `settimeout` on the socket instead of `sleep`.

**Review by StuartFarmer — APPROVED** [review]

**Comment by StuartFarmer** [comment]
addresses

**Pull request merged**